visualime/visualize.py

```diff
@@ -318,6 +319,30 @@ def scale_opacity(
     return new_overlay
 
 
+def scale_overlay(overlay: np.ndarray, shape: Tuple[int, int]) -> np.ndarray:
+    """Scale the overlay to a given size.
+
+    Parameters
+    ----------
+    overlay : np.ndarray
+        The output of :meth:`visualime.visualize.generate_overlay`:
+        An array of shape `(image_width, image_height, 4)` representing an RGBA image.
+
+    shape : tuple of ints
+        The size to scale the overlay to.
+
+    Returns
+    -------
+    np.ndarray
+        An array of shape `(image_width, image_height, 4)` representing an RGBA image.
+
+        Note that this function does not modify the original `ove rlay` but returns a new array.
+    """
+    return np.array(
+        Image.fromarray(overlay.astype(np.uint8)).resize(shape, resample=Image.BICUBIC)
+    )
```

Comment:
Mhh, das ist nicht schlecht, aber dann kann man auch einfach das resultierende Bild skalieren.

Ich dachte, dass man das rein mit den Numpy-Arrays macht, ohne zwischendurch in ein Bild zu konvertieren. Ein bisschen verlustbehaftet skalieren muss man am Ende natürlich immer, aber soweit es geht würde ich das vermeiden wollen.

Siehe auch https://scikit-image.org/docs/stable/auto_examples/transform/plot_rescale.html

Alternativ könnte man auch die Segmente und die Segment-Mask auf dem Bild in Originalgröße skalieren und das Rescaling auf Modellgröße nur in die Predict-Funktion packen. Dann berechnet VisuaLIME alles in Originalgröße und nur bei den Modellaufrufen wird runterskaliert. Das ist natürlich rechnerisch teuer, da bei jedem Modellaufruf skaliert werden muss, aber dürfte insgesamt die beste Qualität erreichen, weil die Maske wirklich pixelgenau auf das Originalbild passt.

Was meinst du?

Reply:
Ich habe es jetzt mit skimage implementiert, damit kann man sich das Konvertieren zu PIL-Bild und wieder zurück zu numpy array sparen